README.md

```diff
@@ -17,11 +38,107 @@ Corpus][corpus], as described by Thorsten Brants and Alex Franz, and [distribute
 Linguistic Data Consortium. Note that this data **"may only be used for linguistic
 education and research"**, so for any other usage you should acquire a different data set.
 
-For the microbenchmark included in this repository, instant-segment is ~17x faster than
+For the microbenchmark included in this repository, Instant Segment is ~17x faster than
 the Python implementation. Further optimizations are planned -- see the [issues][issues].
 The API has been carefully constructed so that multiple segmentations can share
 the underlying state to allow parallel usage.
 
+## Installing
+
+### Python **(>= 3.9)**
+
+```sh
+pip install instant-segment
+```
+
+### Rust
+
+```toml
+[dependencies]
+instant-segment = "*"
+```
+
+## Using
+
+Instant Segment works by segmenting a string into words by selecting the splits with the highest probability given a corpus of words and their occurances.
+
+For instance, provided that `choose` and `spain` occur more frequently than `chooses` and `pain`, Instant Segment can help you split the string `choosespain.com` into [`ChooseSpain.com`](https://instantdomainsearch.com/search/sale?q=choosespain) which more likely matches user intent.
```

Comment:
Hmm, not sure I like this. I like the more subtle callback to IDS in the earlier example, but I'd prefer to be more generic here and just talk about text, leaving domain names out of it (and not linking to IDS like this). On the other hand, would be good to add a sentence or short paragraph somewhere saying explicitly that we built it for IDS, that we are using it in production and what for.

Also, an important part that you don't really call out here is how bigrams are used: that is, the probability of picking `choose spain` over `chooses pain` is not just dependent on unigram frequencies, but also takes into account the frequency of the combination.

Reply:
I do like tying in the domain to IDS since Norvig actually using the domain example choosespain.com in his text.

I'll split this out into something clearer.